pipeline/extract.py

```diff
@@ -0,0 +1,100 @@
+"""Extract script to connect to Bluesky Firehose and extract relevant data given topics"""
+
+import csv
+import ssl
+import json
+import re
+import certifi
+
+from atproto import CAR, models
+from atproto_client.models.utils import get_or_create
+from atproto_firehose import FirehoseSubscribeReposClient, parse_subscribe_repos_message
+
+
+class JSONExtra(json.JSONEncoder):
+    """Serializes raw objects (including CID-Content Identifier) as strings."""
+
+    def default(self, obj: bytes):
+        try:
+            return json.JSONEncoder.default(self, obj)
+        except (TypeError, ValueError, KeyError):
+            return repr(obj)
+
+
+def format_text(text: str) -> str:
+    """Removes extra lines and whitespaces from text"""
+    # Remove consecutive empty lines
+    text = re.sub(r'\n\s*\n', '\n', text)
+
+    text = text.strip()
+    # Replace multiple spaces with a single space
+    text = re.sub(r'\s+', ' ', text)
+
+    return text
+
+
+def extract_text_from_bytes(raw: bytes) -> str:
+    """Extracts text from a raw Bluesky post"""
+    try:
+        json_data = json.dumps(raw, cls=JSONExtra, indent=2)
+        parsed_json = json.loads(json_data)
+        text = parsed_json.get('text')
+        return format_text(text)
+    except (TypeError, AttributeError):
+        return None
```

Comment:
could you add a specific message or make it a bit clearer what exactly is going to happen here?